src/tx/builder/field-types/ct-version.ts

```diff
@@ -35,14 +36,33 @@ export function getProtocolDetails(
 export default {
   serialize(
     value: CtVersion | undefined,
+    params: {},
     { consensusProtocolVersion = ConsensusProtocolVersion.Iris }:
-    { consensusProtocolVersion: ConsensusProtocolVersion },
+    { consensusProtocolVersion?: ConsensusProtocolVersion },
   ): Buffer {
     value ??= getProtocolDetails(consensusProtocolVersion, 'contract-create');
 
     return Buffer.from([value.vmVersion, 0, value.abiVersion]);
   },
 
+  async prepare(
+    value: CtVersion | undefined,
+    params: {},
+    // TODO: { consensusProtocolVersion: ConsensusProtocolVersion } | { onNode: Node } | {}
+    options: { consensusProtocolVersion?: ConsensusProtocolVersion; onNode?: Node },
+  ): Promise<CtVersion | undefined> {
+    if (value != null) return value;
+    if (options.consensusProtocolVersion != null) return undefined;
+    if (Object.keys(ConsensusProtocolVersion).length === 2) return undefined;
```

Comment:
Could '2' be in the constants?

Reply:
`2` means that `ConsensusProtocolVersion` have a single protocol version, e.g.
```
{
  'Iris': 5,
  '5': 'Iris',
}
```
In this case we don't need to find out the current protocol because it won't be compatible with sdk if it is different what we know.
It can be extracted only as `KeysCountInEnumWithASlingleElement`, but I would leave it as a TS quirck.